RecoLocalCalo/EcalDeadChannelRecoveryAlgos/src/EcalDeadChannelRecoveryBDTG.cc

```diff
@@ -0,0 +1,160 @@
+//  Original Authors:  S. Taroni, N. Marinelli
+//  University of Notre Dame - US
+//  Created:
+//
+//
+//
+
+#include "RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryBDTG.h"
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h"  // can I use a egammatools here?
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include <iostream>
+#include <TMath.h>
+
+#include <iostream>
+#include <ostream>
+#include <string>
+#include <fstream>
+
+template <>
+void EcalDeadChannelRecoveryBDTG<EBDetId>::loadFile() {
+  readerNoCrack = new TMVA::Reader("!Color:!Silent");
+
+  for (int i = 0; i < 4; ++i)
```

Comment:
Maybe all this can be simplified (unless order matters: does it?):
```C++
  for (int i = 0; i < 9; ++i) {
    if (i!=4) {
      readerNoCrack->AddVariable(TString("E") + (i + 1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx_.rEn[i]));
      readerCrack->AddVariable(TString("E") + (i + 1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx_.rEn[i]));
    }
    readerNoCrack->AddVariable(TString("iEta") + (i + 1), &(mx_.ieta[i]));
    readerNoCrack->AddVariable(TString("iPhi") + (i + 1), &(mx_.iphi[i]));
    readerCrack->AddVariable(TString("iEta") + (i + 1), &(mx_.ieta[i]));
    readerCrack->AddVariable(TString("iPhi") + (i + 1), &(mx_.iphi[i]));
  }
  readerNoCrack->AddVariable("E1+E2+E3+E4+E6+E7+E8+E9", &(mx_.sumE8));
  readerCrack->AddVariable("E1+E2+E3+E4+E6+E7+E8+E9", &(mx_.sumE8));
```

Reply:
as far as I understood, order matters.

Reply:
Ok, forget my suggestion, then.
(Maybe you can just compactify the first two loops by including the ```if (i!=4)``` clause... Not a big issue anyhow)